docs/TMC_Drivers.md

```diff
@@ -11,18 +11,102 @@ this document are not available.
+However, a higher current also increases the heat produced by the
+stepper motor and the stepper motor driver. If the stepper motor
+driver gets too hot it will disable itself and Klipper will report an
+error. If the stepper motor gets too hot, it loses torque and
+positional accuracy. (If it gets very hot it may also melt plastic
+parts attached to it or near it.)
+
+As a general tuning tip, prefer higher current values as long as the
+stepper motor does not get too hot and the stepper motor driver does
+not report warnings or errors. In general, it is okay for the stepper
+motor to feel warm, but it should not become so hot that it is painful
+to touch.
+
+## Prefer to not specify a hold_current
+
+If one configures a `hold_current` then the TMC driver can reduce
+current to the stepper motor when it detects that the stepper is not
+moving. However, changing motor current may itself introduce motor
+movement. This may occur due to "detent forces" within the stepper
+motor (the permanent magnet in the rotor pulls towards the iron teeth
+in the stator) or due to external forces on the axis carriage.
+
+Most stepper motors will not obtain a significant benefit to reducing
+current during normal prints, because few printing moves will leave a
+stepper motor idle for sufficiently long to activate the
+`hold_current` feature. And, it is unlikely that one would want to
+introduce subtle print artifacts to the few printing moves that do
+leave a stepper idle sufficiently long.
+
+If one wishes to reduce current to motors during print start routines,
+then consider issuing
+[SET_TMC_CURRENT](G-Codes.md#tmc-stepper-drivers) commands in a
+[START_PRINT macro](Slicers.md#klipper-gcode_macro) to adjust the
+current before and after normal printing moves.
+
+Some printers with dedicated Z motors that are idle during normal
+printing moves (no bed_mesh, no bed_tilt, no Z skew_correction, no
+"vase mode" prints, etc.) may find that Z motors do run cooler with a
+`hold_current`. If implementing this then be sure to take into account
+this type of uncommanded Z axis movement during bed leveling, bed
+probing, probe calibration, and similar. The `driver_TPOWERDOWN` and
+`driver_IHOLDDELAY` should also be calibrated accordingly. If unsure,
+prefer to not specify a `hold_current`.
+
+## Setting "spreadCycle" vs "stealthChop" Mode
 
 By default, Klipper places the TMC drivers in "spreadCycle" mode. If
 the driver supports "stealthChop" then it can be enabled by adding
 `stealthchop_threshold: 999999` to the TMC config section.
 
+In general, spreadCycle mode provides greater torque and greater
+positional accuracy than stealthChop mode. Tests comparing modes have
+shown an increase in positional deviation of nearly a full-step when
+using stealthChop mode (for example, on a printer with 40mm
+rotation_distance and 200 steps_per_rotation, position deviation of
+constant speed moves increased by ~0.160mm).
```

Comment:
Very interesting! This seems to be really a lot. And on paper, it looks like the motor is close to skipping a step. So, the question is, is it a random deviation, or a position lag? And if it is a lag, is it velocity- and current- and microstepping-dependent? Any available data would be very helpful. Basically, if the lag comes from internal processing in TMC driver, and not from the loss of torque, it might be a useful recommendation to set the same parameters for the drivers of all moving axes (e.g. same microstepping, and enabling stealthChop for all of them) so that the lag is consistent and reduces de-synchronization between the linear motion of different axes and the extruder, for instance. And if it is due to torque, it indicates, in general, a potentially unsafe operation (but if this lag exists during the constant velocity moves, the resistance to motion should be fairly small, and thus such motion should not require much torque?).

Then a separate question is Z axis, could this introduce a backlash effect and introduce systematic issues with the first layers (and further layers if Z hop or bed meshing is enabled).